### PR TITLE
Force AnimatedVariables to hit begin on 0.0

### LINKED ISSUE #1358: Curves should guarantee that at t=0.0, value=0.0
Once this is fixed, we can remove the additional `_maybeCallOnResized()` call added in #1357.

### PR DESCRIPTION
**Pull request by abarth**
We already forced hitting end on 1.0.

Fixes #1358

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: sky/packages/sky/lib/src/animation/animated_value.dart [hunk 1]
```diff
@@ -57,7 +57,7 @@ class AnimationTiming {
     Interval interval = _getInterval(direction);
     if (interval != null)
       t = interval.transform(t);
-    if (t == 1.0) // Or should we support inverse curves?
+    if (t == 0.0 || t == 1.0) // Or should we support inverse curves?
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/animated_value.dart]: not sure the comment makes sense here

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
Can we assert that this is honoured by all curves somehow?
Also, do we guarantee that t is never < 0.0 or > 1.0, even with floating point errors?

**Comment by abarth** [comment]
I'm not sure what asserts we should add here.  The curve is allowed to go out of [0.0, 1.0] and to switch directions.  We could assert that the end states round to 0.0 and 1.0...

**Comment by abarth** [comment]
> Also, do we guarantee that t is never < 0.0 or > 1.0, even with floating point errors?

Yes, we do actually because at this point `t` is from an `Interval`, not an arbitrary curve.  I've added more asserts to that effect.

**Comment by Hixie** [comment]
> 'm not sure what asserts we should add here

How about adding a constructor to Curve:

```dart
abstract class Curve {
  Curve() {
    assert(transform(0.0) == 0.0);
    assert(transform(1.0) == 1.0);
  }
  double transform(double t);
}
```

**Comment by abarth** [comment]
Unfortunately Curve needs to be const because we use instance as default values of parameters.

Also, the Curves themselves don't hit 0.0 or 1.0.  It's the animation code that calls the curve that enforces that invariant.

**Comment by Hixie** [comment]
Hm, yeah, good point.

I guess this (line 62 in animated_value.dart) is sufficient:
```assert(t == _applyCurve(t, direction).round().toDouble());```

**Pull request merged**